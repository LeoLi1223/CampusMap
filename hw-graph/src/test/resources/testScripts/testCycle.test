# Tests that a graph has a cycle in it, all nodes can be
# printed and all edges can be printed.

# Create a graph
CreateGraph cycle

# Add nodes
AddNode cycle n1
AddNode cycle n2
AddNode cycle n3

# Add edges
AddEdge cycle n1 n2 e12
AddEdge cycle n2 n3 e23
AddEdge cycle n3 n1 e31

# Print all nodes and edges
ListNodes cycle
ListChildren cycle n1
ListChildren cycle n2
ListChildren cycle n3