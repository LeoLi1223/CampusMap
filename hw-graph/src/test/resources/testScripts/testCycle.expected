# Tests that a graph has a cycle in it, all nodes can be
# printed and all edges can be printed.

# Create a graph
created graph cycle

# Add nodes
added node n1 to cycle
added node n2 to cycle
added node n3 to cycle

# Add edges
added edge e12 from n1 to n2 in cycle
added edge e23 from n2 to n3 in cycle
added edge e31 from n3 to n1 in cycle

# Print all nodes and edges
cycle contains: n1 n2 n3
the children of n1 in cycle are: n2(e12)
the children of n2 in cycle are: n3(e23)
the children of n3 in cycle are: n1(e31)